cpp/src/arrow/compute/api_vector.cc

```diff
@@ -151,6 +152,10 @@ static auto kRankOptionsType = GetFunctionOptionsType<RankOptions>(
     DataMember("sort_keys", &RankOptions::sort_keys),
     DataMember("null_placement", &RankOptions::null_placement),
     DataMember("tiebreaker", &RankOptions::tiebreaker));
+static auto kRankQuantileOptionsType = GetFunctionOptionsType<RankQuantileOptions>(
+    DataMember("sort_keys", &RankQuantileOptions::sort_keys),
+    DataMember("null_placement", &RankQuantileOptions::null_placement),
+    DataMember("factor", &RankQuantileOptions::factor));
```

Comment:
Since we do have time maybe its worth removing the factor to simplify? What is the advantage of including into the QuantileRanker class versus return something in the range 0..1 and having the user supply the factor if they need? 

I figure the advantage to taking the factor out would be to simplify the implementation and give more room for vectorized multiplication applications

Reply:
@zanmato1984 @cyb70289 @icexelloss  Do you think the factor would be useful to keep?

Reply:
I don't see it being too useful - I think most of the cases (0, 1) is enough

Reply:
I don't have enough experience of using functions alike so I'll respect others' opinions.

Reply:
Ok, I removed the `factor` option.